File: app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/EventDao.java
package de.tum.in.tca.ticketcheck.component.ticket;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

import de.tum.in.tca.ticketcheck.component.ticket.model.Event;

@Dao
public interface EventDao {

    /**
     * Removes all old items
     */
    @Query("DELETE FROM event WHERE date < date('now')")
    void removePastEvents();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(List<Event> event);

    @Query("SELECT * FROM event ORDER BY date")
    List<Event> getAll();

    @Query("SELECT * FROM event where id = :id")
    Event getEventById(int id);

    @Query("DELETE FROM event")
    void removeAll();

}

File: app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/EventDao.kt
package de.tum.`in`.tca.ticketcheck.component.ticket

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

import de.tum.`in`.tca.ticketcheck.component.ticket.model.Event

@Dao
interface EventDao {

    @Query("SELECT * FROM event ORDER BY date")
    fun getAll(): List<Event>

    @Query("SELECT * FROM event where id = :id")
    fun getEventById(id: Int): Event

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(event: List<Event>)

    @Query("DELETE FROM event WHERE date < date('now')")
    fun removePastEvents()

}
